cylc/flow/task_events_mgr.py

```diff
@@ -777,24 +787,23 @@ def _process_message_check(
             return False
 
         if (
-                itask.state(TASK_STATUS_WAITING)
-                and
+            itask.state(TASK_STATUS_WAITING)
+            and itask.tdef.run_mode == 'live'   # Polling in live mode only.
```

Comment:
The code within this section is designed to avoid problems with polling - but there will be no polling in simulation mode.